public/src/managers/TempDataProvider.ts

```diff
@@ -187,9 +187,11 @@ export class TempDataProvider implements IUserProvider, ICourseProvider {
         throw new Error("Method not implemented");
     }
 
-    public async changeUserState(link: Enrollment, state: Enrollment.UserStatus): Promise<boolean> {
+    public async changeUserState(link: Enrollment, state: Enrollment.UserStatus): Promise<Status> {
         link.setStatus(state);
-        return true;
+        const stat = new Status();
+        stat.setCode(0);
+        return stat;
```

Comment:
Can this be written as: `return new Status().setCode(0);`  or `return (new Status()).setCode(0);`  ?

Reply:
Unfortunately, not. Both `new Status().setCode(0)` and `(new Status()).setCode(0);` expressions are functions returning a void, we cannot return them when a promise of type `Status` is expected.